crates/gosub_instance/src/lib.rs

```diff
@@ -0,0 +1,198 @@
+use gosub_interface::chrome::ChromeHandle;
+use gosub_interface::config::{HasTreeDrawer, ModuleConfiguration};
+use gosub_interface::draw::TreeDrawer;
+use gosub_interface::eventloop::EventLoopHandle;
+use gosub_interface::instance::{Handles, InstanceId};
+use gosub_interface::render_backend::{ImageBuffer, NodeDesc};
+use gosub_shared::geo::SizeU32;
+use gosub_shared::types::Result;
+use std::sync::mpsc::Sender as SyncSender;
+use tokio::runtime::{Builder, Runtime};
+use tokio::sync::mpsc::{Receiver, Sender};
+use tokio::task;
+use tokio::task::{spawn_blocking, LocalSet};
+use url::Url;
+
+pub struct EngineInstance<C: ModuleConfiguration> {
```

Comment:
I don't know exactly what an engine instance is... but from context I don't understand why it needs a title or url?

Reply:
It is basically a tab. It will manage the event loop, the web event loop and everything else.

It needs the title and url, so JS can ask for it